### DISCUSSION TITLE
Overriding Email Settings

### DISCUSSION THREAD

**Discussion by mikesnoeren** [craftcms/cms#10591] [Ideas] [answered]
### Description
When overriding the default email settings via `/app/config/app.php` craft will show an error when pressing the "Test' button on the Email Settings page: `There was an error testing your email settings.`

I suspect this is happening because the 'Test' button runs the test with the values on that page and not with the overridden values? If that's the case, there should be an indicator to tell the user about this behaviour.. 
Not having an indicator telling me what is up made me believe the error was on my side.
It wasn't untill I tried the cli command `craft mailer/test` that I found out the error wasn't on my end, as the test runs fine that way.. (shouldn't those two behaviours be the same anyways?)

Another thing I noticed on this page was that all settings are able to be overriden by environment variables, except the Transport Type setting. Why can't that be set to an environment variable?

My `app.php`:
```php
    'dev' => [
        'components' => [
            'mailer' => function () {
                $settings = craft\helpers\App::mailSettings();

                if (App::env('MAILHOG_HOST')) {
                    $settings->transportType = craft\mail\transportadapters\Smtp::class;
                    $settings->transportSettings = [
                        'host'              => App::env('MAILHOG_HOST'),
                        'port'              => '1025',
                        'useAuthentication' => 0,
                    ];
                }

                $config = craft\helpers\App::mailerConfig($settings);
                return Craft::createObject($config);
            }
        ],
    ],
```

### Steps to reproduce

1. use app.php to overrule mail settings
2. test mail settings via craft > Settings >Email Settings > Test

### Additional info

- Craft version: 3.7.32
- PHP version: 7.3
- Database driver & version: mysql:5.7
- Plugins & versions: -

**Comment by brandonkelly** [comment, marked as answer]
This is working as intended for now. The “Test” button will test the settings that are directly in front of you on the settings page.
